api/rest/client/methods_test.go

```diff
@@ -161,6 +161,52 @@ func TestUnpin(t *testing.T) {
 	testClients(t, api, testF)
 }
 
+func TestPinPath(t *testing.T) {
+	api := testAPI(t)
+	defer shutdown(api)
+
+	testF := func(t *testing.T, c Client) {
+		testPaths := test.Paths()
+
+		for _, testPath := range testPaths {
```

Comment:
This is better, but you should define here in the test, a `type testcase struct` which includes the `path` you're testing with and the expected response (and whether an error is expected or not). When a test fails, I'd like to have the values we're testing with in sight.

Also, `t.Error` messages should give more information about the problem, since in case of failure we only get information about the line. So it's better to log the current testcase failing.